shell/common/common_TPC-H.sh

```diff
@@ -1,40 +1,73 @@
 # Common functions for different TPC-H implementations
 # Based on Hadoop and Hive
-
-source_file "$ALOJA_REPO_PATH/shell/common/common_hive.sh"
-set_hive_requires
-
-[ ! "$TPCH_SCALE_FACTOR" ] &&  TPCH_SCALE_FACTOR=1 #1 GB min size
-[ ! "$TPCH_USE_LOCAL_FACTOR" ] && TPCH_USE_LOCAL_FACTOR="" #set to a scale factor to use the local DBGEN instead of the M/R version
-
-BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
-
+#
+if [[ "$NATIVE_FORMAT" == "text" && "$BENCH_SUITE" == *"native-spark"* ]]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hadoop.sh"
+  set_hadoop_requires
+else
+  source_file "$ALOJA_REPO_PATH/shell/common/common_hive.sh"
+  set_hive_requires
+fi
+
+if [ "$BB_SERVER_DERBY" == "true" ]; then
+  source_file "$ALOJA_REPO_PATH/shell/common/common_derby.sh"
+  set_derby_requires
+fi
+
+
+if [[ "$TPCH_USE_LOCAL_FACTOR" > 0 ]]; then
+  TPCH_SCALE_FACTOR=$TPCH_USE_LOCAL_FACTOR
+  BENCH_DATA_SIZE="$((TPCH_USE_LOCAL_FACTOR * 1000000000 ))" #in bytes
+else
+  [[ ! "$TPCH_SCALE_FACTOR" ]] && TPCH_SCALE_FACTOR=1 #1 GB min size
+  BENCH_DATA_SIZE="$((TPCH_SCALE_FACTOR * 1000000000 ))" #in bytes
+fi
 
 TPCH_HDFS_DIR="/tmp/tpch-generate"
-TPCH_DB_NAME="tpch_${BENCH_FILE_FORMAT}_${TPCH_SCALE_FACTOR}"
 
-[ ! "$BENCH_LIST" ] && BENCH_LIST="$(seq -f "tpch_query%g" -s " " 1 22)"
+if [[ ! "$NATIVE_FORMAT" == "text"  || ! "$BENCH_SUITE" == *"native-spark"* ]]; then
+  [ ! -z $NATIVE_FORMAT ] && BENCH_FILE_FORMAT=$NATIVE_FORMAT
+  TPCH_DB_NAME="tpch_${BENCH_FILE_FORMAT}_${TPCH_SCALE_FACTOR}"
+fi
+
+[ ! "$BENCH_LIST" ] && BENCH_LIST="$(printf '%d ' {1..22})"
```

Comment:
So `BENCH_LIST` went from being:

```
tpch_query1
tpch_query2
...
```

to:
```
1 2 3 4
```

Is that right?

Reply:
Yes, it was necessary to maintain consistency between TPC-H on native spark and D2F benchmarks. They used different names so option -l didn't work. Using query number only respects other approaches like BB and permits leverage the logic of finding the query file to the bench itself.